packages/api/cms-api/src/page-tree/page-tree-read-api.ts

```diff
@@ -94,17 +98,31 @@ export function createReadApi(
             if (where.slug) {
                 nodes = nodes.filter((node) => node.slug === where.slug);
             }
+            if (sort) {
+                sort.map((sortItem) => {
+                    nodes = nodes.sort((a, b) => {
+                        if (sortItem.field === "updatedAt") {
+                            return sortItem.direction === "ASC"
+                                ? new Date(a.updatedAt).valueOf() - new Date(b.updatedAt).valueOf()
+                                : new Date(b.updatedAt).valueOf() - new Date(a.updatedAt).valueOf();
+                        } else {
+                            return sortItem.direction === "ASC" ? a[sortItem.field] - b[sortItem.field] : b[sortItem.field] - a[sortItem.field];
+                        }
+                    });
+                });
+            }
+            if (offset !== undefined && limit !== undefined) {
+                nodes = nodes.slice(offset, limit);
+            }
+
             return nodes;
         } else {
             return tracer.startActiveSpan("live query PageTree queryNodes", async (span) => {
                 span.setAttribute("scope", JSON.stringify(scope));
                 span.setAttribute("where", JSON.stringify(where));
-                const qb = pageTreeNodeRepository
-                    .createQueryBuilder()
-                    .where({
-                        visibility: visibilityFilter,
-                    })
-                    .orderBy({ pos: "ASC" });
```

Comment:
you could leave the (most common) orderBy pos and only sort in JS when the order differs

Reply:
how would this work with the paging parameters? the order needs to be taken into account when applying offset and limit?

the only case where this would make sense is when all nodes have been preloaded to only sort when the order differs, as the preloaded nodes are ordered by position anyway. is this what you meant?

Reply:
Now the default sort for `pos: ASC` is removed from the given sorting array when its the first item and only the remaining items are used